metrics/integration_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Metrics Integration", Ordered, func() {
+	BeforeAll(func() {
+		metrics.Registry.Unregister(SnapshotRunningSeconds)
+	})
+
+	var (
+		AttemptSnapshotRunningSecondsHeader = inputHeader{
+			Name: "integration_snapshot_running_seconds",
+			Help: "Integration durations from the moment the buildPipelineRun is completed/ snapshot resource was created til the snapshot is marked as in progress status",
+		}
+	)
+
+	Context("When RegisterRunningSnapshot is called", func() {
```

Comment:
Maybe instead "Context when" just `When` can be called
```suggestion
	When("RegisterRunningSnapshot is called", func() {
```

Reply:
More context will be added later. So just keep it now.